plugins/flexibleengine/src/main/java/org/eclipse/xpanse/plugins/flexibleengine/common/FlexibleEngineClient.java

```diff
@@ -0,0 +1,176 @@
+
+import com.huaweicloud.sdk.ces.v1.CesClient;
+import com.huaweicloud.sdk.core.HcClient;
+import com.huaweicloud.sdk.core.HttpListener;
+import com.huaweicloud.sdk.core.SdkResponse;
+import com.huaweicloud.sdk.core.auth.BasicCredentials;
+import com.huaweicloud.sdk.core.auth.ICredential;
+import com.huaweicloud.sdk.core.exception.ServiceResponseException;
+import com.huaweicloud.sdk.core.http.HttpConfig;
+import com.huaweicloud.sdk.core.region.Region;
+import com.huaweicloud.sdk.ecs.v2.EcsClient;
+import com.huaweicloud.sdk.iam.v3.IamClient;
+import com.huaweicloud.sdk.iam.v3.model.KeystoneListProjectsRequest;
+import com.huaweicloud.sdk.iam.v3.model.KeystoneListProjectsResponse;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.xpanse.modules.models.monitor.exceptions.ClientApiCallFailedException;
+import org.springframework.stereotype.Component;
+
+/**
+ * FlexibleEngine Service Client.
+ */
+@Slf4j
+@Component
+public class FlexibleEngineClient extends FlexibleEngineCredentials {
+
+    /**
+     * Get client for service ECS.
+     *
+     * @param credential credential
+     * @param regionName region name
+     * @return client for service ECS.
+     */
+    public EcsClient getEcsClient(ICredential credential, String regionName) {
+        HcClient hcClient = getHcClient(credential, ECS_ENDPOINT_PREFIX, regionName);
+        return new EcsClient(hcClient);
+    }
+
+    /**
+     * Get client for service CES.
+     *
+     * @param credential credential
+     * @param regionName region name
+     * @return client for service CES.
+     */
+    public CesClient getCesClient(ICredential credential, String regionName) {
+        HcClient hcClient = getHcClient(credential, CES_ENDPOINT_PREFIX, regionName);
+        return new CesClient(hcClient);
+    }
+
+    /**
+     * Match retry condition.
+     *
+     * @param response response
+     * @param ex       exception
+     * @return true if match retry condition, otherwise false
+     */
+    public boolean matchRetryCondition(SdkResponse response, Exception ex) {
+        if (Objects.isNull(ex)) {
+            return false;
+        }
+        if (!ServiceResponseException.class.isAssignableFrom(ex.getClass())) {
+            return false;
+        }
+        int statusCode = ((ServiceResponseException) ex).getHttpStatusCode();
+        return statusCode == ERROR_CODE_TOO_MANY_REQUESTS
+                || statusCode == ERROR_CODE_INTERNAL_SERVER_ERROR;
+    }
+
+    private HcClient getHcClient(ICredential credential, String servicePrefix, String regionName) {
+        String endpoint = PROTOCOL_HTTPS + servicePrefix + regionName + ENDPOINT_SUFFIX;
+        Region region = new Region(regionName, endpoint);
```

Comment:
this region variable is not used anywhere.

Reply:
Removed.